examples/chat-app/messages.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"context"
+	"math/big"
+	"sync"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+	"github.com/libp2p/go-smart-record/ir"
+	"github.com/libp2p/go-smart-record/protocol"
+)
+
+type clientConfig struct {
```

Comment:
As we talked earlier, modularize/simplify the implementation by implementing a seprate io.ReadCloser and io.WriteCloser that use an SR client under the hood.